### DISCUSSION TITLE
Best database engine on aws

### DISCUSSION THREAD

**Discussion by ngocnn97** [DefectDojo/django-DefectDojo#8836] [Q&A] [answered]
I am planning to deploy Defect-Dojo on AWS, let me ask which Database Engine I should choose between the following types:

- Aurora MySQL 
- Aurora PostgreSQL 
- RDS MySQL
- RDS PostgreSQL

According to the [documentation](https://defectdojo.github.io/django-DefectDojo/getting_started/architecture/), should I choose RDS MySQL?

**Comment by manuel-sommer** [comment]
I would suggest you use Aurora PostgreSQL, but it depends also on the amount of findings you upload. According to documentation, PostgreSQL is faster in writing compared with MySQL, but MySQL is faster in reading. In Aurora you have better options to scale the database and have it more stable. I used RDS MySQL in the past, but I will now try out Aurora PostgreSQL. What is your opinion @quirinziessler ?

**Comment by quirinziessler** [comment]
I am currently running RDS Mysql. Now having about 200k findings and aggregating those, page speed dropped to 9 seconds of dashboard loading time and users starting complaining about it. So I wouldn't suggest to use it if you start from scratch. Also I have some warnings with the MyISAM tables I am just ignoring right now.

Aurora may be helpful, as it uses some more features in terms of scalability and speed. As AWS states, speed is x3 compared to RDS PostgreSQL and x5 RDS MySQL. In case I would now start from scratch and considering the speed enhancement by Aurora I would choose Aurora MYSQL if it supports MyISAM tables like the MySQL RDS. Otherwise Aurora PostgreSQL.

Please let us know which you choose and if there are any issues. Maybe we could improve this part of the docs a bit.

**Reply by ngocnn97** [reply]
RDS Mysql support MyIASM so why you have some warnings ?

**Reply by quirinziessler** [reply]
Yeah it supports MyISAM but seems to have problems with re-storing those tables. I didn't tried it out so far to be honest so I can't tell you if this may be a FP warning. Converting btw shouldn't be done as these MyISAM tables are required by django-watson.

Here is the warning:
DB Instance XYZ contains MyISAM tables that have not been migrated to InnoDB. These tables can impact your ability to perform point-in-time restores. Consider converting these tables to InnoDB. Please refer to http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.BackingUpAndRestoringAmazonRDSInstances.html#Overview.BackupDeviceRestrictions

**Comment by ngocnn97** [comment]
I'm using Aurora MySQL with about 80k findings, can't load the Dashboard page, I advise you not to use it, I think it's because Aurora MySQL doesn't support MyISAM.
I'm wondering whether to switch to Aurora PostgreSQL or (RDS MySQL because it supports MyISAM, watson-django needs MyIASM)

**Reply by quirinziessler** [reply]
unhappy to hear that, but thanks for trying and sharing. I would suggest you to go with Aurora PostgreSQL and take use of the scalability and maybe also caching features.

**Reply by ngocnn97** [reply, marked as answer]
Thanks you, i will try Aurora PostgreSQL

**Comment by manuel-sommer** [comment]
Do you know @ngocnn97 if it is possible to separate write and read through Aurora endpoints and configure it with DefectDojo? 
Do you use docker-compose or k8s?

**Reply by ngocnn97** [reply]
i am using k8s, I never thought to separate reading and writing into two Aurora endpoints.

**Reply by manuel-sommer** [reply]
https://github.com/DefectDojo/django-DefectDojo/issues/8871
